dashboard/src/components/Common/XpubField/XpubField.jsx

```diff
@@ -23,11 +28,27 @@ class XpubField extends React.Component {
         this.props.didLoadAutocomplete()
       })
     }
+
+    this.props.typeProps.onChange(Object.keys(methodOptions)[0])
   }
 
   render() {
-    const radioChanged = event => {
-      this.setState({ selectedType: event.target.value })
+    const {
+      typeProps,
+      valueProps,
+      mockhsmKeys,
+    } = this.props
+
+    const typeOnChange = event => {
+      const value = typeProps.onChange(event).value
+
+      typeProps.onChange(value)
```

Comment:
Can you explain what's going on here? It looks like we're calling `typeProps.onChange()` twice, and passing its return value from the first call as an argument to the second call.